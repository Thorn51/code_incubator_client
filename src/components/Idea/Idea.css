.idea_page_header {
  display: flex;
  justify-content: center;
  height: 40vh;
  background: linear-gradient(
        45deg,
        rgba(39, 38, 1, 0.8),
        rgba(0, 35, 170, 0.7),
        rgba(53, 49, 21, 0.7),
        rgba(248, 211, 0, 0.7)
      )
      0 0 / 1000% no-repeat,
    url("https://images.unsplash.com/photo-1566148252861-3def7b71c74d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60");
  background-size: cover;
  background-position: center center;
  background-attachment: fixed;
  margin: 0px;
}

.project_details,
.idea_description {
  background-color: white;
  padding: 15px;

  border-radius: 10px;
  box-shadow: 3px 3px #84828f;
}
.app_name {
  font-size: 2em;
  color: white;
  text-emphasis-color: black;
}

.project_vote {
  background-color: #011627;
}

.bold {
  font-weight: bolder;
}

@media only screen and (min-width: 600px) {
  .app_name {
    font-size: 3em;
  }
}

@media only screen and (min-width: 1224px) {
  .app_name {
    font-size: 4em;
  }
}
