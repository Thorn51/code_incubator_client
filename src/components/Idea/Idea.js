import React from "react";
import "./Idea.css";
import ThumbDownIcon from "@material-ui/icons/ThumbDown";
import ThumbUpIcon from "@material-ui/icons/ThumbUp";
import moment from "moment";

export default function Idea(props) {
  return (
    <>
      <header className="idea_page_header">
        <h1 className="app_name">{props.idea.project_title}</h1>
        <div className="project_vote">
          <ThumbUpIcon onClick={() => props.projectUpVote(props.idea.id)}>
            thumb_up
          </ThumbUpIcon>
          <p className="votes_for">{props.idea.votes}</p>
          <ThumbDownIcon onClick={() => props.projectDownVote(props.idea.id)}>
            thumb_down
          </ThumbDownIcon>
        </div>
      </header>
      <section className="project_details">
        <span className="project_author">
          Submitted by: {props.author.nickname}
        </span>
        <span className="date_submitted">
          Submitted: {moment(props.idea.date_submitted).format("MMM Do YYY")}
        </span>
        <span className="project_status">Status: {props.idea.status}</span>
        <span className="repo">Repo: {props.idea.github}</span>
      </section>

      <section className="idea_description">
        <h2 className="section_title">Overview</h2>
        <p>{props.idea.project_summary}</p>
      </section>
    </>
  );
}

Idea.defaultProps = {
  idea: {},
  author: {}
};
