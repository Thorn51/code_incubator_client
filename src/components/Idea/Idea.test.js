import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Idea from "./Idea";

describe("<Idea />", () => {
  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Idea />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Renders Idea as expected", () => {
    const tree = renderer.create(<Idea />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
