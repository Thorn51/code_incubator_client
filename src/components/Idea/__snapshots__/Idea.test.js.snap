// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Idea /> Renders Idea as expected 1`] = `
Array [
  <header
    className="idea_page_header"
  >
    <h1
      className="app_name"
    />
    <div
      className="project_vote"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        onClick={[Function]}
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M1 21h4V9H1v12zm22-11c0-1.1-.9-2-2-2h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2z"
        />
      </svg>
      <p
        className="votes_for"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        onClick={[Function]}
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 3H6c-.83 0-1.54.5-1.84 1.22l-3.02 7.05c-.09.23-.14.47-.14.73v2c0 1.1.9 2 2 2h6.31l-.95 4.57-.03.32c0 .41.17.79.44 1.06L9.83 23l6.59-6.59c.36-.36.58-.86.58-1.41V5c0-1.1-.9-2-2-2zm4 0v12h4V3h-4z"
        />
      </svg>
    </div>
  </header>,
  <section
    className="project_details"
  >
    <span
      className="project_author"
    >
      Submitted by: 
    </span>
    <span
      className="date_submitted"
    >
      Submitted: 
    </span>
    <span
      className="project_status"
    >
      Status: 
    </span>
    <span
      className="repo"
    >
      Repo: 
    </span>
  </section>,
  <section
    className="idea_description"
  >
    <h2
      className="section_title"
    >
      Overview
    </h2>
    <p />
  </section>,
]
`;
